yt_idv/scene_data/block_collection.py

```diff
@@ -41,6 +41,39 @@ def add_data(self, field, no_ghost=False):
         # Every time we change our data source, we wipe all existing ones.
         # We now set up our vertices into our current data source.
         vert, dx, le, re = [], [], [], []
+
+        # spherical-only: SHOULD ONLY DO THIS IF DATASET IS SPHERICAL
+        # normal vectors and plane constants for min/max phi face
+        phi_plane_le = []
+        phi_plane_re = []
+
+        axis_id = self.data_source.ds.coordinates.axis_id
+
+        def calculate_phi_normal_plane(le_or_re):
```

Comment:
do we spend enough time in this function that it makes sense to vectorize it?  i.e., call if on a list of coordinates and then return an array?

Reply:
Vectorizing this is a good idea! and I think will simplify un-tangling the spherical-only part of the calculation from the base cartesian version.

Reply:
this is vectorized now! see https://github.com/yt-project/yt_idv/pull/64/files#diff-7d5c480ce667d255f45fb76e3578809a9840831967712edc900858d049cc5517